### PR TITLE
Update to use Hibernate ORM 5.1.3.Final

### PR DESCRIPTION
**Pull request by kabir**
https://issues.jboss.org/browse/WFLY-6854
Replaces #9413

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/SFSBHibernate2LcacheStats.java [hunk 1]
```diff
@@ -79,6 +79,7 @@ public void setupConfig() {
             Configuration configuration = new Configuration().setProperty(AvailableSettings.USE_NEW_ID_GENERATOR_MAPPINGS,
                     "true");
             configuration.getProperties().put(AvailableSettings.JTA_PLATFORM, JBossAppServerJtaPlatform.class);
+            configuration.getProperties().put(AvailableSettings.TRANSACTION_COORDINATOR_STRATEGY, "jta");
```
Comment (kabir) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/SFSBHibernate2LcacheStats.java]: Is this a breaking change? Why do they need this now? In the worst case I guess they can document it?

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/secondlevelcache/SFSB.java [hunk 2]
```diff
@@ -70,9 +70,9 @@ public void setupConfig() {
         try {
 
             // prepare the configuration
-            Configuration configuration = new Configuration().setProperty(AvailableSettings.USE_NEW_ID_GENERATOR_MAPPINGS,
-                    "true");
+            Configuration configuration = new Configuration().setProperty(AvailableSettings.USE_NEW_ID_GENERATOR_MAPPINGS, "true");
             configuration.getProperties().put(AvailableSettings.JTA_PLATFORM, JBossAppServerJtaPlatform.class);
+            configuration.getProperties().put(AvailableSettings.TRANSACTION_COORDINATOR_STRATEGY, "jta");
```
Comment (kabir) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/secondlevelcache/SFSB.java]: Why is this needed now?

**Review by kabir — COMMENTED** [review]
@scottmarlow @gbadner I have squashed the commits from the original PR #9413 to this one. If you are around by any chance, can you please provide feedback.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/SFSBHibernate2LcacheStats.java (hunk 1 above)
Comment (scottmarlow) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/SFSBHibernate2LcacheStats.java]: https://paste.fedoraproject.org/491994/03435341 contains the failure that we get without the AvailableSettings.TRANSACTION_COORDINATOR_STRATEGY setting.  I need to find the related emails where we discussed this, as I don't want to go memory. :)

Comment (scottmarlow) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/SFSBHibernate2LcacheStats.java]: tl;dr

https://hibernate.atlassian.net/browse/HHH-11024?focusedCommentId=82236&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-82236 

I need to find the above referenced prior conversation.

https://github.com/hibernate/hibernate-orm/blob/5.0/migration-guide.adoc#transactions mentions the change for ORM 5.0/5.1.  We are currently using Hibernate ORM 5.0.10 with EAP 7.0

https://issues.jboss.org/browse/JBEAP-5768 (EAP doc change jira for "Non-JPA applications that use Hibernate with JTA must set hibernate.transaction.coordinator_class") mentions that the ORM 5.0 doc change was via commit https://github.com/hibernate/hibernate-orm/commit/f23b550b623a484dd56ccd1028fdaee34c64f907.

Comment (scottmarlow) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/SFSBHibernate2LcacheStats.java]: ORM 5.0.x added the TRANSACTION_COORDINATOR_STRATEGY="jta" setting requirement but it wasn't until later in ORM 5.1, that a bug fix (HHH-11024) required this to be set, that impacts schema drop code.

> git annotate ./hibernate-core/src/main/java/org/hibernate/cfg/AvailableSettings.java | grep hibernate.transaction.coordinator_class
> 7990630a	(Steve Ebersole	2015-07-27 17:05:41 -0500	184)	String TRANSACTION_COORDINATOR_STRATEGY = "hibernate.transaction.coordinator_class";

The summary is that a Hibernate ORM bug was fixed in 5.1, that depends on the application correctly specifying TRANSACTION_COORDINATOR_STRATEGY="jta".

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/secondlevelcache/SFSB.java (hunk 2 above)
Comment (scottmarlow) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/secondlevelcache/SFSB.java]: ORM 5.0.x added the TRANSACTION_COORDINATOR_STRATEGY="jta" setting requirement but it wasn't until later in ORM 5.1, that a bug fix (HHH-11024) required this to be set, that impacts schema drop code.  So, its not a new setting, as ORM 5.0.x also had the setting.
